src/awx_plugins/credentials/azure_kv.py

```diff
@@ -71,20 +86,41 @@
     ],
     'required': [
         'url',
-        'client',
-        'secret',
-        'tenant',
         'secret_field',
     ],
 }
 
 
-def azure_keyvault_backend(**kwargs):
-    csc = ClientSecretCredential(
-        tenant_id=kwargs['tenant'],
-        client_id=kwargs['client'],
-        client_secret=kwargs['secret'],
-    )
+def azure_keyvault_backend(**kwargs: str) -> str:
+    """Get a credential and retrieve a secret from an Azure Key Vault.
+
+    :param url: An Azure Key Vault URI.
+    :param client (optional): The Client ID.
+    :param secret (optional): The Client Secret.
+    :param tenant (optional): The Tenant ID.
+    :param secret_field: The name of the secret to retrieve from the
+        vault.
+    :param secret_version (optional): The version of the secret to
+        retrieve.
+    :returns: The secret from the Key Vault.
+    :raises: RuntimeError: If the software is not being run on an Azure
+        VM.
+    """
+    try:
+        if '' in [kwargs['tenant'], kwargs['client'], kwargs['secret']]:
```

Comment:
Which of these lines is expected to raise an exception? It's an antipattern to have more than one instruction in a try-block.

Reply:
https://github.com/ansible/awx-plugins/blob/fdfe0f6df76715ead11b9b7fc25181b56cf0f163/src/awx_plugins/credentials/azure_kv.py#L111 will raise the exception if the code is not run on an Azure VM but we should only try to use ManagedIdentityCredential if one of the variables in that list is missing